cdc/sorter/leveldb/db/leveldb.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/sorter"
+	"github.com/pingcap/ticdc/pkg/config"
+	cerrors "github.com/pingcap/ticdc/pkg/errors"
+	"github.com/pingcap/ticdc/pkg/retry"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"go.uber.org/zap"
+)
+
+// OpenDB opens a leveldb.
+func OpenDB(ctx context.Context, id int, cfg *config.SorterConfig) (DB, error) {
+	lcfg := cfg.LevelDB
+	var option opt.Options
+	option.OpenFilesCacheCapacity = lcfg.MaxOpenFiles / lcfg.Count
```

Comment:
Perhaps it would be better to initialize these parameters by abstracting them into a function?

Reply:
Do you mean builder pattern? `Builder().SetA().SetB().Build()`